scarlet/component.py

```diff
@@ -267,22 +274,13 @@ def K(self):
         return self.n_components
 
     @property
-    def B(self):
-        """Number of bands
-        """
-        return self.components[0].B
-
-    @property
-    def Ny(self):
-        """Number of pixels in the y-direction
+    def frame(self):
+        """Frame of the components.
         """
-        return np.max([comp.Ny for comp in self.components])
-
-    @property
-    def Nx(self):
-        """Number of pixels in the y-direction
-        """
-        return np.max([comp.Nx for comp in self.components])
+        try:
+            return self._frame
+        except AttributeError:
+            return self.components[0].frame
```

Comment:
If the `ComponentTree` does not have a `_frame` attribute then I would actually institute a check here to ensure that all of the components have the same frame , then set `_frame` to that frame.

Reply:
True, but it's a somewhat contrived scenario. The only reason this isn't merely a `return self._frame` is that we would initialize `Blend.frame` as a `ComponentTree` from the sources alone. `Blend` wouldn't need the `frame` argument at all. This is a change of dependency logic, but I'm open for discussion.

We currently say `frame` is the frame for Blend as a model. So, it should better have `frame` as argument.

But we can also say, `frame` is the frame for the sources/components as the fundamental units of the model, and `Blend` is just their collection. Then `Blend` doesn't need `frame`.

Reply:
I disagree. I think that all of the components need to have the same `Frame`, otherwise `ComponentTree.get_model` doesn't work. It also breaks the interchangeability of `Component` and `ComponentTree`, which depends on `ComponentTree` having all of the same attributes and methods as `Component` (plus additional methods needed for the parent) in order to allow sources/components to be hierarchical. 

So the tree seems to be the natural place to define the `Frame` for all of its components and check that they all agree. Really this should probably be done in `ComponentTree.__init__` and `Component.frame` should just return `self._frame`. We should also check that two components/trees have equal frames in `__iadd__`, and implement a few new tests to make sure that initialization and addition fail if they don't.

Reply:
(but the tests can be done in a separate PR)

Reply:
The consistency check is not a problem, and a good idea.

But the sources need `frame` to get initialized, the question is does `Blend`? If we decide, we treat it in the tree, which makes sense to me, too, `Blend` doesn't need it, and we should remove it now while we're making API changes.